ENIGMAsystem/SHELL/Platforms/SDL/Joystick.cpp

```diff
@@ -79,20 +79,28 @@ namespace enigma {
 bool joystick_init() {
   double init;
   init = (SDL_InitSubSystem(SDL_INIT_JOYSTICK) > 0);
-  joystick1 = SDL_JoystickOpen(0);
-  joystick2 = SDL_JoystickOpen(1);
+  if (joystick1 == NULL)
+    joystick1 = SDL_JoystickOpen(0);
+  
+  if (joystick2 == NULL)
+    joystick2 = SDL_JoystickOpen(1);
+  
   return init;
 }
 
 void joystick_uninit() {
-  SDL_JoystickClose(joystick1);
-  SDL_JoystickClose(joystick2);
+  if (joystick1 != NULL)
+    SDL_JoystickClose(joystick1);
+  
+  if (joystick2 != NULL)
+    SDL_JoystickClose(joystick2);
+  
   SDL_QuitSubSystem(SDL_INIT_JOYSTICK);
 }
   
 void joystick_update() {  
   int joystick_count = SDL_NumJoysticks();
-  if (joystick_count == 0) {
+  if (joystick_count <= 0) {
```

Comment:
This logic is basically identical to `joystick_uninit`. Please move it there and call it here.